components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/PasswordGrantHandler.java

```diff
@@ -174,6 +200,86 @@ private boolean validateUserCredentials(OAuth2AccessTokenReqDTO tokenReq, Servic
         return true;
     }
 
+    /**
+     * This method will publish the Password Grant Authentication data.
+     *
+     * @param tokenReq Token request which contains all the details of the request.
+     * @param authenticated Boolean value which determines whether the user is authenticated or not.
+     * @param serviceProvider Service provider which contains the details of the application.
+     */
+    private void publishAuthenticationData(OAuth2AccessTokenReqDTO tokenReq, boolean authenticated,
```

Comment:
Shall we make this method protected?
Will be useful for anyone extending the grant handler.

Reply:
Modified.